lib/double_entry/reporting/line_metadata_filter.rb

```diff
@@ -0,0 +1,33 @@
+# encoding: utf-8
+module DoubleEntry
+  module Reporting
+    class LineMetadataFilter
+
+      def self.filter(collection:, metadata:)
+        table_alias_index = 0
+
+        metadata.reduce(collection) do |filtered_collection, (key, value)|
+          table_alias = "m#{table_alias_index}"
+          table_alias_index += 1
+
+          filtered_collection.
+            joins("INNER JOIN #{line_metadata_table} as #{table_alias} ON #{table_alias}.line_id = #{lines_table}.id").
```

Comment:
If anyone knows of a way to do this without having to write out the SQL I am all :ear:s! I couldn't figure out a way to do it though, especially as we will accept any number of key/value pairs.